whisper.cpp

```diff
@@ -439,6 +442,11 @@ struct whisper_context {
 template<typename T>
 static void read_safe(std::ifstream& fin, T& dest) {
     fin.read((char*)& dest, sizeof(T));
+#if defined(GGML_BIG_ENDIAN)
+    if constexpr (std::endian::native == std::endian::big) {
```

Comment:
I realise the reason for the guard is because `<bit>` is guarded by it, but this seems wrong; if that's defined, it will always be BE.

What you probably want is something like `#if __cplusplus >= 202110L` (the revision of C++2x that added `std::byteswap`).